controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/Deployment.java

```diff
@@ -46,6 +48,7 @@ public Deployment(ZoneId zone, ApplicationVersion applicationVersion, Version ve
         this.clusterInfo = Objects.requireNonNull(clusterInfo, "clusterInfo cannot be null");
         this.metrics = Objects.requireNonNull(metrics, "deploymentMetrics cannot be null");
         this.activity = Objects.requireNonNull(activity, "activity cannot be null");
+        this.loadBalancers = Objects.requireNonNull(loadBalancers, "loadBalancers cannot be null");
```

Comment:
Please wrap this with `ImmutableMap#copyOf`. The same should be done for the `clusterUtilization` and `clusterInfo` fields.